src/firestore.js

```diff
@@ -0,0 +1,25 @@
+import { call } from 'redux-saga/effects'
+
+function * getDoc (collectionRef, docRef) {
+  const doc = this._getCollection(collectionRef, 'firestore').doc(docRef)
+  const result = yield call([doc, doc.get])
+
+  return result.data()
+}
+
+function * getCollection (collectionRef) {
+  const collection = this._getCollection(collectionRef, 'firestore')
+  const querySnapshot = yield call([collection, collection.get])
+  let result
+
+  querySnapshot.forEach((doc) => {
```

Comment:
I'm not sure we should do this kind of transformation in the lib, it removes control from the user: they lose access to the [firebase.firestore.QuerySnapshot API](https://firebase.google.com/docs/reference/js/firebase.firestore.QuerySnapshot).

I'd rather simply return the `querySnapshot` directly.

Reply:
What if we do both?

I would like to have a single method I can reliably get document data, so my frontend components simply have to map props, instead of worrying about transforming the DocumentReference.

There are other times, perhaps developing better offline data management, where accessing the metadata would be important.

I see a need for both methods and was leaning that direction for the other pull request. Thoughts?

Reply:
yes @KamesCG i think that we can  do both...can you add it in a another PR? For now i return the querySnapshot directly.